### DISCUSSION TITLE
Export PDF as Images with Custom Page Selection in SumatraPDF

### DISCUSSION THREAD

**Discussion by DharaniDJ** [sumatrapdfreader/sumatrapdf#4616] [Q&A]
Hello @kjk! I'm looking to export PDFs as images (PNG or JPEG) in SumatraPDF, ideally with an option for users to select custom page ranges. Does this feature currently exist in the application? If not, could anyone guide me on which files or components in the source code would be the best starting points to implement this feature? Any help would be greatly appreciated!

Thank you!

**Comment by GitHubRulesOK** [comment]
There are many ways that can be done but not native to Windows or MuPDF **viewer**
HOWEVER is a MuPDF Mutool ability that could be called.

The main problem with such commands is the desire to specify a "range" and that then makes it niche.

To export all file pages as PDF or PNG or JPG is no different to "burst by printing"

Rather than virtual GhostScript-ed print pages by range  (which would be the easy generic MS Windows option) 
We can specify any app to do the page splitting by send to ExternalViewer and ask the user to define their desired changes or ranges

This is the basis of my send pages to booklet printing 

or save as PNG here https://github.com/GitHubRulesOK/MyNotes/blob/master/AppNotes/SumatraPDF/Addins/ExportPng/ExportPng.cmd

However that can be simplified to be more graceful and less native I have just updated to self download the last common 32bit mutool.exe
![image](https://github.com/user-attachments/assets/6b094e73-4d26-4d6d-859b-7789e9925d42)
```
	[
		CommandLine = "C:\Users\K\AppData\Local\SumatraPDF\exportpng.cmd" "%1" page=1-N
		Name = Export all pages to Png
		Filter = *.pdf
		Key = P
	]
```
![image](https://github.com/user-attachments/assets/e62990a6-bdf3-4bdb-a262-134220c52427)


so works as a system "drag and drop" a file onto cmd or sendto here I asked for page 5 of an epub.
![image](https://github.com/user-attachments/assets/35aea5d2-7d6d-4750-adeb-2dce9322c342)

It should be easily adaptable if other filetypes are later supported 
![image](https://github.com/user-attachments/assets/cbe1383c-812d-4186-a5d6-07a4abd296ea)
